nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/flow-designer/ui/canvas/items/flow/flow-diff-dialog/flow-diff-dialog.ts

```diff
@@ -0,0 +1,298 @@
+ * limitations under the License.
+ */
+
+import {
+    AfterViewInit,
+    Component,
+    DestroyRef,
+    ViewChild,
+    inject
+} from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogModule } from '@angular/material/dialog';
+import { MatTableDataSource, MatTableModule } from '@angular/material/table';
+import { MatSort, MatSortModule } from '@angular/material/sort';
+import { FormBuilder, FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { CommonModule } from '@angular/common';
+import { MatFormField, MatLabel } from '@angular/material/form-field';
+import { MatSelectModule } from '@angular/material/select';
+import { MatOption } from '@angular/material/core';
+import { MatInput } from '@angular/material/input';
+import { MatButton } from '@angular/material/button';
+import { MatProgressSpinner } from '@angular/material/progress-spinner';
+import { combineLatest, of } from 'rxjs';
+import { catchError, debounceTime, distinctUntilChanged, filter, map, startWith, switchMap, take } from 'rxjs/operators';
+import { FlowComparisonEntity, VersionControlInformation } from '../../../../../state/flow';
+import { VersionedFlowSnapshotMetadata } from '../../../../../../../state/shared';
+import { RegistryService } from '../../../../../service/registry.service';
+import { CloseOnEscapeDialog, NiFiCommon } from '@nifi/shared';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { Store } from '@ngrx/store';
+import { CanvasState } from '../../../../../state';
+import { selectTimeOffset } from '../../../../../../../state/flow-configuration/flow-configuration.selectors';
+
+export interface FlowDiffDialogData {
+    versionControlInformation: VersionControlInformation;
+    versions: VersionedFlowSnapshotMetadata[];
+    currentVersion: string;
+    selectedVersion: string;
+}
+
+interface FlowDiffRow {
+    componentName: string;
+    changeType: string;
+    difference: string;
+}
+
+@Component({
+    selector: 'flow-diff-dialog',
+    standalone: true,
+    imports: [
+        CommonModule,
+        MatDialogModule,
+        MatTableModule,
+        MatSortModule,
+        ReactiveFormsModule,
+        MatFormField,
+        MatLabel,
+        MatSelectModule,
+        MatOption,
+        MatInput,
+        MatButton,
+        MatProgressSpinner
+    ],
+    templateUrl: './flow-diff-dialog.html',
+    styleUrl: './flow-diff-dialog.scss'
+})
+export class FlowDiffDialog extends CloseOnEscapeDialog implements AfterViewInit {
+    private data = inject<FlowDiffDialogData>(MAT_DIALOG_DATA);
+    private registryService = inject(RegistryService);
+    private destroyRef = inject(DestroyRef);
+    private formBuilder = inject(FormBuilder);
+    private nifiCommon = inject(NiFiCommon);
+    private store = inject<Store<CanvasState>>(Store);
+    private timeOffset = this.store.selectSignal(selectTimeOffset);
+
+    @ViewChild(MatSort)
```

Comment:
You can follow how the change version dialog implemented sorting instead of using a ViewChild here.